Dockerfile

```
@@ -0,0 +1,17 @@
+FROM python:3.9.6-slim-buster as base 
+
+RUN apt-get update \
+    && apt-get -y install curl \
+    && curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+
+COPY . /app 
+WORKDIR /app 
+ENV PATH="${PATH}:/root/.poetry/bin"
+RUN poetry install 
```

Currently Docker will have to rebuild starting from line 7 whenever any file is changed.

If you instead split things up like the following, then Docker will skip the poetry install step when dependencies haven't changed:
```suggestion
WORKDIR /app
ENV PATH="${PATH}:/root/.poetry/bin"
COPY pyproject.toml poetry.lock ./
RUN poetry install
COPY . ./
```